HealthcareSystem/Backend/Service/SendingMail/MailService.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Backend.Model.Users;
+using Backend.Settings;
+using MailKit.Net.Smtp;
+using MailKit.Security;
+using Microsoft.Extensions.Options;
+using MimeKit;
+
+namespace Backend.Service.SendingMail
+{
+    public class MailService : IMailService
+    {
+        private readonly MailSettings _mailSettings;
+
+        public MailService(IOptions<MailSettings> mailSettings)
+        {
+            _mailSettings = mailSettings.Value;
+        }
+        public async Task SendWelcomeEmailAsync(WelcomeRequest request)
+        {
+            string FilePath = Directory.GetCurrentDirectory() + "\\Templates\\WelcomeTemplate.html";
+            StreamReader str = new StreamReader(FilePath);
+            string MailText = str.ReadToEnd();
+            str.Close();
+            MailText = MailText.Replace("[username]", request.UserName).Replace("[email]", request.ToEmail).Replace("[jmbg]", request.Jmbg);
+            var email = new MimeMessage();
+            email.Sender = MailboxAddress.Parse(_mailSettings.Mail);
+            email.To.Add(MailboxAddress.Parse(request.ToEmail));
+            email.Subject = $"Welcome {request.UserName}";
+            var builder = new BodyBuilder();
+            builder.HtmlBody = MailText;
+            email.Body = builder.ToMessageBody();
+            using var smtp = new SmtpClient();
+            smtp.Connect(_mailSettings.Host, _mailSettings.Port, SecureSocketOptions.StartTls);
+            smtp.Authenticate(_mailSettings.Mail, _mailSettings.Password);
+            await smtp.SendAsync(email);
+            smtp.Disconnect(true);
+        }
```

Comment:
This method is too long, we could do some refactoring. 

### My proposal for refactoring this class.

```sh
using System.IO;
using System.Threading.Tasks;
using Backend.Model.Users;
using Backend.Settings;
using MailKit.Net.Smtp;
using MailKit.Security;
using Microsoft.Extensions.Options;
using MimeKit;

namespace Backend.Service.SendingMail
{
    public class MailService : IMailService
    {
        private readonly MailSettings _mailSettings;

        public MailService(IOptions<MailSettings> mailSettings)
        {
            _mailSettings = mailSettings.Value;
        }
        public async Task SendWelcomeEmailAsync(WelcomeRequest request)
        {
            var MailText = ParseMailText(request, ReadMailText());
            var email = ConfigureEmail(request, MailText);
            await SendEmail(email);
        }

        private async Task SendEmail(MimeMessage email)
        {
            using var smtp = new SmtpClient();
            smtp.Connect(_mailSettings.Host, _mailSettings.Port, SecureSocketOptions.StartTls);
            smtp.Authenticate(_mailSettings.Mail, _mailSettings.Password);
            await smtp.SendAsync(email);
            smtp.Disconnect(true);
        }

        private MimeMessage ConfigureEmail(WelcomeRequest request, string MailText)
        {
            var email = new MimeMessage();
            email.Sender = MailboxAddress.Parse(_mailSettings.Mail);
            email.To.Add(MailboxAddress.Parse(request.ToEmail));
            email.Subject = $"Welcome {request.UserName}";

            var builder = new BodyBuilder();
            builder.HtmlBody = MailText;
            email.Body = builder.ToMessageBody();
            return email;
        }

        private string ParseMailText(WelcomeRequest request, string MailText)
        {
            var parsedMailText = MailText.Replace("[username]", request.UserName).Replace("[email]", request.ToEmail)
                .Replace("[jmbg]", request.Jmbg);
            return parsedMailText; ;
        }

        private string ReadMailText()
        {
            var pathToHtmlPage = "\\Templates\\WelcomeTemplate.html";
            string FilePath = Directory.GetCurrentDirectory() + pathToHtmlPage;
            StreamReader reader = new StreamReader(FilePath);
            string MailText = reader.ReadToEnd();
            reader.Close();
            return MailText;
        }
    }
}


```

Reply:
Applied as suggested in f3dd33a7ec46560896650fb783d63778066c2722.